openapi_python_client/parser/openapi.py

```diff
@@ -266,6 +273,20 @@ def _add_parameters(
 
         return endpoint, schemas
 
+    @staticmethod
+    def _sort_parameters(*, endpoint: "Endpoint", path: str) -> Union["Endpoint", ParseError]:
+        endpoint = deepcopy(endpoint)
+        parameters_form_path = re.findall(_PATH_PARAM_REGEX, path)
+        path_parameter_names = [p.name for p in endpoint.path_parameters]
+        if sorted(parameters_form_path) != sorted(path_parameter_names):
+            return ParseError(
+                data=endpoint.path_parameters,
+                detail="Incorrect path templating (Path parameters do not match with path)",
+            )
+
+        endpoint.path_parameters.sort(key=lambda p: parameters_form_path.index(p.name))
```

Comment:
```suggestion
        parameters_from_path = re.findall(_PATH_PARAM_REGEX, path)
        try:
            endpoint.path_parameters.sort(key=lambda p: parameters_from_path.index(p.name))
        except ValueError:
            pass  # We're going to catch the difference down below
        path_parameter_names = [p.name for p in endpoint.path_parameters]
        if parameters_from_path != path_parameter_names:
            return ParseError(
                data=endpoint.path_parameters,
                detail="Incorrect path templating (Path parameters do not match with path)",
            )
```

I believe this will do the same thing with two fewer sorting calls. Also I changed form -> from since I believe that was the intention.